sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs

```diff
@@ -0,0 +1,1413 @@
+            options ??= new MapsWeatherClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _pipeline = HttpPipelineBuilder.Build(options, new MapsSasCredentialPolicy(credential));
+            restClient = new WeatherRestClient(_clientDiagnostics, _pipeline, endpoint, null, options.Version);
+        }
+
+        /// <summary> Initializes a new instance of MapsWeatherClient. </summary>
+        /// <param name="clientDiagnostics"> The handler for diagnostic messaging in the client. </param>
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="clientId"> Specifies which account is intended for usage in conjunction with the Microsoft Entra ID security model.  It represents a unique ID for the Azure Maps account and can be retrieved from the Azure Maps management  plane Account API. To use Microsoft Entra ID security in Azure Maps see the following <see href="https://aka.ms/amauthdetails">articles</see> for guidance. </param>
+        /// <param name="apiVersion"> Api Version. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/> or <paramref name="pipeline"/> is null. </exception>
+        internal MapsWeatherClient(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri endpoint = null, string clientId = null, MapsWeatherClientOptions.ServiceVersion apiVersion = MapsWeatherClientOptions.LatestVersion)
+        {
+            var options = new MapsWeatherClientOptions(apiVersion);
+            restClient = new WeatherRestClient(clientDiagnostics, pipeline, endpoint, clientId, options.Version);
+            _clientDiagnostics = clientDiagnostics;
+            _pipeline = pipeline;
+        }
+
+        /// <summary>
+        /// [Protocol Method] Use to get a detailed hourly weather forecast for up to 24 hours or a daily forecast for up to 10 days.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// This <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md">protocol method</see> allows explicit creation of the request and processing of the response for advanced scenarios.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="options"> Get hourly forecast options to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        public virtual async Task<Response<HourlyForecastResult>> GetHourlyForecastAsync(GetHourlyForecastOptions options, CancellationToken cancellationToken = default)
```

Comment:
Can we rename it to `GetHourlyWeatherForecastAsync` and `GetHourlyWeatherForecast` ? It's not very clear to me at the first glance of what hourly forecast these methods want to forecast.

Reply:
Same for `GetMinuteForecastAsync`, `GetMinuteForecast`, `GetQuarterDayForecastAsync`, `GetQuarterDayForecast`